internal/ratelimit/handler.go

```diff
@@ -114,10 +116,29 @@ func extractResourceAction(path, method string) (res, act string, err error) {
 // set to the number of seconds the client should wait to make it's next request.
 func Handler(l *rate.Limiter, next http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		path := req.URL.Path
-		method := req.Method
+		reqInfo, ok := event.RequestInfoFromContext(req.Context())
+		if !ok || reqInfo == nil || reqInfo.ClientIp == "" {
+			// TODO: what to do here?
+			// We could:
+			// 1. Ignore and continue ahead, with the ip being empty...
+			// 2. Have fallback logic to extract the Ip
+			// 3. Return an error since it would mean the middlewares are not setup correctly.
+			rw.WriteHeader(http.StatusInternalServerError)
```

Comment:
I agree with option 3.

Reply:
To be clear, by "return an error" I mean what is currently implemented here with setting a 500 response header and returning from the handler.